cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Context.java

```diff
@@ -203,69 +205,126 @@ public PropertyBags phaseTapChangerTable(String tableId) {
-        }
+        handleIssue(ConversionIssueCategory.IGNORED, what, reason);
     }
 
     public void pending(String what, Supplier<String> reason) {
-        if (LOG.isInfoEnabled()) {
-            LOG.info("PENDING {}. Reason: {}", what, reason.get());
-        }
+        handleIssue(ConversionIssueCategory.PENDING, what, reason);
     }
 
     public void fixed(String what, String reason) {
-        LOG.warn(FIXED_REASON, what, reason);
-    }
-
-    public void fixed(Supplier<String> what, String reason) {
-        if (LOG.isWarnEnabled()) {
-            LOG.warn(FIXED_REASON, what.get(), reason);
-        }
+        handleIssue(ConversionIssueCategory.FIXED, what, reason);
     }
 
     public void fixed(String what, Supplier<String> reason) {
-        if (LOG.isWarnEnabled()) {
-            LOG.warn(FIXED_REASON, what, reason.get());
-        }
+        handleIssue(ConversionIssueCategory.FIXED, what, reason);
     }
 
     public void fixed(String what, String reason, double wrong, double fixed) {
-        LOG.warn("Fixed {}. Reason: {}. Wrong {}, fixed {}", what, reason, wrong, fixed);
+        Supplier<String> reason1 = () -> String.format("%s. Wrong %.4f, was fixed to %.4f", reason, wrong, fixed);
+        handleIssue(ConversionIssueCategory.FIXED, what, reason1);
     }
 
     public void missing(String what) {
-        LOG.warn("Missing {}", what);
+        String reason1 = "";
+        handleIssue(ConversionIssueCategory.MISSING, what, reason1);
     }
 
     public void missing(String what, double defaultValue) {
-        LOG.warn("Missing {}. Used default value {}", what, defaultValue);
+        Supplier<String> reason1 = () -> String.format("Using default value %.4f", defaultValue);
+        handleIssue(ConversionIssueCategory.MISSING, what, reason1);
+    }
+
+    private void handleIssue(ConversionIssueCategory category, String what, String reason) {
+        handleIssue(category, what, () -> reason);
+    }
+
+    private void handleIssue(ConversionIssueCategory category, String what, Supplier<String> reason) {
+        reportIssue(category, what, reason);
+        logIssue(category, what, reason);
+    }
+
+    private static void logIssue(ConversionIssueCategory category, String what, Supplier<String> reason) {
+        if (LOG.isWarnEnabled()) {
+            LOG.warn("{}: {}. Reason: {}", category, what, reason.get());
+        }
+    }
+
+    private static boolean isValidForGrouping(String what) {
+        return what.equals("SvVoltage") || what.equals("Regulating Terminal") || what.equals("EmptyVscRegulation");
+    }
+
+    private void reportIssue(ConversionIssueCategory category, String what, Supplier<String> reason) {
+        Reporter categoryReporter = issueReporters.computeIfAbsent(category, cat -> reporter.createSubReporter(cat.name(), cat.toString()));
+        // FIXME(Luma) A first approach to functional logs, report issues grouped by "what" or by "reason"
```

Comment:
I'm not completely convinced we should do this grouping, or at least that way. This could be done by the functional logs viewer, but the strength of what you proposed is if there are many similar logs. Then you might want to show to the user a group of similar issues instead of the details of all issues, and the details can be looked into if he wants by unfolding this group.

Nonetheless, the title of that subReporter should then sum up the content, for instance the number of issues of that type, which is not the case here. There's some refactoring needed to do that.